feat: add `$app/state` module (#13140)
sveltejs/kit: documentation/docs/30-advanced/67-shallow-routing.md

	}
</script>

{#if page.state.showModal}
	<Modal close={() => history.back()} />
{/if}
```

The modal can be dismissed by navigating back (unsetting `page.state.showModal`) or by interacting with it in a way that causes the `close` callback to run, which will navigate back programmatically.

## API

The first argument to `pushState` is the URL, relative to the current URL. To stay on the current URL, use `''`.

> The second argument is the new page state, which can be accessed via the [page state]($app-state#page) as `page.state`. You can make page state type-safe by declaring an [`App.PageState`](types#PageState) interface (usually in `src/app.d.ts`).

Author: 'via the page state as page.state' is why we can't really call `page` 'page state', we should refer to it as 'the page object' instead

```suggestion
The second argument is the new page state, which can be accessed via the [page object]($app-state#page) as `page.state`. You can make page state type-safe by declaring an [`App.PageState`](types#PageState) interface (usually in `src/app.d.ts`).
```

To set page state without creating a new history entry, use `replaceState` instead of `pushState`.

> [!LEGACY]

> > `page.state` from `$app/state` exists since SvelteKit 2.12. If you're using an earlier version or are using Svelte 4, use `$page.state` from `$app/stores` instead.

Author: ```suggestion
> `page.state` from `$app/state` was added in SvelteKit 2.12. If you're using an earlier version or are using Svelte 4, use `$page.state` from `$app/stores` instead.
```
